### PR TITLE
Add InferenceLog proto

### PR DESCRIPTION
**Pull request by terryyylim**
**What this PR does / why we need it**:

This PR exposes InferenceLog proto to be used by Merlin's logger, that is to be open-sourced in the coming weeks.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by krithika369** [comment]
@terryyylim I think the Python 3.7 related CI jobs are probably failing due to some stale cache. Could you try clearing those from the runner?

File: Makefile [hunk 1]
```diff
@@ -205,6 +205,13 @@ generate-proto:
 		--go-json_out=../../api \
 		--go-json_opt=module=github.com/gojek/merlin \
 		transformer/**/*.proto
+	@cd protos/merlin && \
+		protoc -I=. \
+		--go_out=../../api \
+		--go_opt=module=github.com/gojek/merlin \
+		--go-json_out=../../api \
+		--go-json_opt=module=github.com/gojek/merlin \
+		log/*.proto
```
Comment (krithika369) [inline Makefile:214]: Can the 2 commands to generate the protos be combined into one?

File: api/pkg/log/inference_log.pb.json.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
+// Code generated by protoc-gen-go-json. DO NOT EDIT.
```
Comment (krithika369) [inline api/pkg/log/inference_log.pb.json.go:1]: How and why was this file generated?

Comment (terryyylim) [reply api/pkg/log/inference_log.pb.json.go:1]: Its generated via the `protoc` command:
```
--go-json_out=../../api \
--go-json_opt=module=github.com/gojek/merlin \
```

**Review by krithika369 — APPROVED** [review]
LGTM. Thanks for this @terryyylim !

**Pull request merged**